--- qtsoap/qtsoap.pro.install-pub-headers
+++ qtsoap/qtsoap.pro
@@ -3,3 +3,9 @@
 include(common.pri)
 qtsoap-uselib:SUBDIRS=buildlib
-SUBDIRS+=examples
+SUBDIRS+=examples
+INSTALLS += PUB_HEADERS
+
+PUB_HEADERS.files = src/QtSoapArray src/QtSoapHttpTransport src/QtSoapNamespaces src/QtSoapSimpleType src/QtSoapStructIterator \
+                    src/QtSoapTypeConstructor src/QtSoapTypeFactory src/QtSoapArrayIterator src/QtSoapMessage src/QtSoapQName \
+                    src/QtSoapStruct src/QtSoapType src/QtSoapTypeConstructorBase src/qtsoap.h
+PUB_HEADERS.path = $$[QT_INSTALL_HEADERS]/QtSoap
